client/human/HumanClientApp.h

```diff
@@ -72,7 +70,12 @@ class HumanClientApp :
     void                ChangeLoggerThreshold(const std::string& option_name, LogLevel option_value);
 
     void                ResetToIntro(bool skip_savegame);
-    void                ExitApp();
+
+    /** Exit the App with \p exit_code. */
+    void                Exit(int exit_code) override;
+    /** Exit SDL. (Called by FSM) */
```

Comment:
some explanations here about why these are separate functions and/or why you would use one or the other in particular situations would be helpful...

Reply:
I have refactored the code so that `GG::GUI::Exit()` is called `GG::GUI::ExitApp()`, `ExitSDL()` is private and the FSM is passed a `std::function<void()>` as an action to take after the server has exited.  Hopefully, being private makes it clear that users of `HumanClientApp` should not call `ExitSDL()` 

A better solution would have been to directly bind `SDLGUI::ExitApp()` in `ResetOrExitApp()`, so that there is no need to expose the private one liner `ExitSDL()`.  However, `SDLGUI` is a pure virtual class and all forms of intentional slicing result in uncompileable code.